static/js/settings_display.js

```diff
@@ -14,6 +14,25 @@ const meta = {
     loaded: false,
 };
 
+function get_default_language_name(language_list, language_code) {
+    const language_list_map = {};
+
+    // One-to-one mapping from code to name for all languages
+    for (const language of language_list) {
+        language_list_map[language.code] = language.name;
+    }
+    return language_list_map[language_code];
+}
+
+export let default_language_name = get_default_language_name(
+    page_params.language_list,
```

Comment:
Why does `language_list` need to be a parameter here to this function?  It seems like `get_default_language_name` can just use it.  

Also, we generally aim to avoid calling functions at import time; this should be called from an `initialize` method in `ui_init.js`.  I think looking at patterns there may suggest further cleanups.